slides/lesson1/solution/main_hard_code.tf

```diff
@@ -58,17 +58,23 @@ resource "aws_instance" "web" {
     command = "bash -c 'MAX=10; C=0; until curl -s -o /dev/null ${aws_instance.web.public_dns}; do [ $C -eq $MAX ] && { exit 1; } || sleep 10; ((C++)); done;' || false"
   }
 
-  user_data = "${data.template_file.user_data.rendered}"
+  user_data = data.template_file.user_data.rendered
 
-  vpc_security_group_ids = ["${aws_security_group.web.id}"]
+  vpc_security_group_ids = [aws_security_group.web.id]
 
-  tags = "${merge(var.tags, map("Name", format("%s-%s",data.aws_caller_identity.current.user_id,"ec2")))}"
+  tags = merge(
+    var.tags,
+    {
+      "Name" = format("%s-%s", data.aws_caller_identity.current.user_id, "ec2")
+    },
+  )
 }
 
 resource "aws_route53_record" "dns_web" {
```

Comment:
Do we expect the participants to already have a hosted zone in route53? My recommendation would be to keep the DNS-related resources commented out

Reply:
this is an internal record, not an external one.